pandas/core/categorical.py

```diff
@@ -1735,31 +1737,34 @@ def convert_categorical(x):
             return x.get_values()
         return x.ravel()
 
-    typs = get_dtype_kinds(to_concat)
-    if not len(typs-set(['object','category'])):
-
-        # we only can deal with object & category types
-        pass
-
-    else:
-
+    if get_dtype_kinds(to_concat) - set(['object', 'category']):
```

Comment:
ok, back at computer.

I think the issue is what do do with an ``object`` that has same or < categories (then ok), but what if it has more categories (uniques).

So I would do this.

have all matching categories (exactly), with no object in a single path
else just push everything to object.